src/libraries/System.Text.Json/tests/Serialization/CyclicTests.cs

```diff
@@ -57,7 +58,7 @@ public static void WriteCyclicFail(int objectHierarchyDepth, int maxDepth, int e
             // Since the last Parent property is null, the serializer moves onto the Children property.
             string expectedPath = "$" + string.Concat(Enumerable.Repeat(".Parent", expectedPathDepth));
             Assert.Contains(expectedPath, ex.Path);
-            Assert.Contains(maxDepth.ToString(), ex.ToString());
+            Assert.Contains(effectiveMaxDepth.ToString(), ex.ToString());
```

Comment:
```suggestion
            Assert.Contains(effectiveMaxDepth.ToString(), ex.Message);
```

`ex.ToString()` also contains the call stack (including line numbers), which could give us false positives.

Reply:
Yeah, I thought about this, but didn't want to change the existing test too much. I'll do a more specific check.

Reply:
I amended the commit